Fix validation refusal detection

AdrienClairembault opened the pull request:
Internal ref: 20749.

Consider the following situation:
1) Minimum approval for a given ticket is 100%.
2) There is 10 validators.

If 1 validator refuses the approval, the ticket is still "pending for validation" despite not being able to be validated even if all the 9 remaining validators votes yes.

This is because we use a very simple formula for checking refusal:
`{refusal %} > {minimum_validation %}`

In this case, it translates to 10% > 100% which is false -> no action.

---

These changes implement the following formula:
`{refusal %} + {minimum_validation %} > 100%`

In the case above, it translates to (10% + 100%) > 100% which is true -> approval is denied.



| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

orthagh reviewed (commented):
By the way, a dedicated test for this case would be appreciated

cedric-anne commented on inc/commonitilvalidation.class.php:
```diff
@@ -1405,7 +1405,7 @@ static function computeValidationStatus(CommonITILObject $item) {
          if (($statuses[self::ACCEPTED]*100/$total) >= $validation_percent) {
             $validation_status = self::ACCEPTED;
-         } else if (($statuses[self::REFUSED]*100/$total) >= $validation_percent) {
+         } else if (($statuses[self::REFUSED]*100/$total) + $validation_percent > 100) {
```
Maybe it would b more comprehensible to write `{refusal %} > 100% - {minimum_validation %}` (i.e. `10% > 100% - 100%`.
```suggestion
         } else if (($statuses[self::REFUSED]*100/$total) > 100 - $validation_percent) {
```

AdrienClairembault commented:
> By the way, a dedicated test for this case would be appreciated

I've moved the logic to another method to make it unit testable and added a test.

  
> Maybe it would b more comprehensible to write {refusal %} > 100% - {minimum_validation %} (i.e. 10% > 100% - 100

I find it less readable this way, IMO it's easier to read a calculation on the left side of a condition than on the right.
